src/blueteam/Archon.java

```diff
@@ -39,23 +39,26 @@ void step() throws GameActionException {
 		}
 
 		// we want to preserve some bullet points for gardener
-		if (!canSpent() && getRobotCount(RobotType.GARDENER) > 0) {
+		if (!canSpent() && getRobotCount(RobotType.GARDENER) > 1) {
 			return;
 		}
 
 		// randomly attempt to build a gardener if we need more
 		Direction dir = randomDirection();
-		int numOfGardeners = getRobotCount(RobotType.GARDENER);
-		int gardenersNeeded = TeamConstants.DESIRED_NUMBER_OF_GARDENERS - numOfGardeners;
+		System.out.println(getRobotCount(RobotType.GARDENER));
 		if (rc.canHireGardener(dir)
-				&& Math.random() < (double) gardenersNeeded / numberOfArchons) {
+				&& TeamConstants.DESIRED_NUMBER_OF_GARDENERS > getRobotCount(RobotType.GARDENER)) {
```

Comment:
Naive way, just check whether there is enough gardeners. Works well in one archon scenario.